qiskit/providers/ibmq/runtime/ibm_runtime_service.py

```diff
@@ -406,16 +406,34 @@ def set_program_visibility(self, program_id: str, public: bool) -> None:
                 If ``False``, make the program visible to just your account.
 
         Raises:
-            RuntimeJobNotFound: if program not found (404)
+            RuntimeProgramNotFound: if program not found (404)
             QiskitRuntimeError: if update failed (401, 403)
         """
         try:
             self._api_client.set_program_visibility(program_id, public)
         except RequestsApiError as ex:
             if ex.status_code == 404:
-                raise RuntimeJobNotFound(f"Program not found: {ex.message}") from None
+                raise RuntimeProgramNotFound(f"Program not found: {ex.message}") from None
             raise QiskitRuntimeError(f"Failed to set program visibility: {ex}") from None
 
+    def update_program(self, program_id: str, program_str: str) -> None:
```

Comment:
We have to provide a similar interface to upload_program. See [this tutorial](https://github.com/Qiskit-Partners/qiskit-runtime/blob/main/tutorials/02_uploading_program.ipynb) on how a user uploads a program from an external python file. So `program_str: str` should instead be `data: Union[bytes, str]`. If data is a str then it means user is providing a file path, else user is passing program data as bytes.

Reply:
Updated! Support is now [here](https://github.com/Qiskit/qiskit-ibmq-provider/pull/1008/files#diff-d9a771d60c46a277056063a1949d6e3e0191fd2770bf6fc69c49c78dcf4c38a0R222) and the new integration testing is now [here](https://github.com/Qiskit/qiskit-ibmq-provider/pull/1008/files#diff-68d8af3bb045bfa53f3f7c127891b2276392f8797f0d3a768c151f6a2f41869cR158).